src/rdf4cpp/query/Variable.cpp

```diff
@@ -3,9 +3,53 @@
 #include <rdf4cpp/InvalidNode.hpp>
 #include <rdf4cpp/util/CharMatcher.hpp>
 #include <rdf4cpp/writer/TryWrite.hpp>
+
 #include <uni_algo/all.h>
 
+#include <cstring>
+
 namespace rdf4cpp::query {
+
+namespace detail_variable_inlining {
+    inline constexpr size_t max_inlined_name_len = (storage::identifier::NodeID::width / (sizeof(char) * 8)) - 1; // -1 for anonymous tagging boolean
+
+    struct InlinedRepr {
+        char name[max_inlined_name_len];
+        bool is_anonymous;
```

Comment:
could we store the bool in the highest bit of the last char? that should always be 0 for ASCII
(and if it is set, disable inlining for that variable)

Reply:
probably, but I don't think its worth the effort/code complexity just to get one more inlined char